File: problems/basic/232/solution.py
# Accepted: 35ms (73.14%), 17.56MB (13.10%)

class MyStack:
    def __init__(self):
        self.stack = []

    def push(self, value: int) -> None:
        self.stack.append(value)

    def pop(self) -> int:
        return self.stack.pop()

    def peek(self) -> int:
        return self.stack[-1]

    def size(self) -> int:
        return len(self.stack)

class MyQueue:
    def __init__(self):
        self.stack1 = MyStack()
        self.stack2 = MyStack()

    def push(self, x: int) -> None:
        self.stack1.push(x)

    def pop(self) -> int:
        if self.stack2.size() == 0:
            while self.stack1.size() > 0:
                self.stack2.push(self.stack1.pop())
        
        return self.stack2.pop()

    def peek(self) -> int:
        if self.stack2.size() == 0:
            while self.stack1.size() > 0:
                self.stack2.push(self.stack1.pop())
        
        return self.stack2.peek()

    def empty(self) -> bool:
        return self.stack1.size() == 0 and self.stack2.size() == 0


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: problems/basic/232/solution.cpp
// Accepted: 4ms (6.77%), 7.54MB (14.56%)

class MyStack {
    vector<int> stack;
public:
    void push(int value) {
        stack.push_back(value);
    }

    int pop() {
        int pop = stack.back();
        stack.pop_back();
        return pop;
    }

    int peek() {
        return stack.back();
    }

    int size() {
        return stack.size();
    }
};

class MyQueue {
    MyStack stack1;
    MyStack stack2;
public:
    void push(int x) {
       stack1.push(x); 
    }

    int pop() {
        if (stack2.size() == 0) {
            while (stack1.size() > 0) {
                stack2.push(stack1.pop());
            }
        }

        return stack2.pop();
    }

    int peek() {
        if (stack2.size() == 0) {
            while (stack1.size() > 0) {
                stack2.push(stack1.pop());
            }
        }

        return stack2.peek();
    }

    bool empty() {
        return stack1.size() == 0 && stack2.size() == 0;
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */
